ui/src/actions/handleResizeImage.ts

```diff
@@ -0,0 +1,38 @@
+import { ImageDimensions } from "@/types/imageDimensions";
+import axios from "axios";
+import { FormEvent } from "react";
+import toast from "react-hot-toast";
+
+const handleResizeImage = (e: FormEvent, newDimensions: ImageDimensions, filename: string) => {
+  e.preventDefault();
+  
+  const {width, height} = newDimensions;
+
+  if(!width || !height) {
+    toast.error("Width and height are required!")
+    return;
+  }
+
+  const data = {
+    filename,
+    width: Math.abs(Math.floor(width)),
+    height: Math.abs(Math.floor(height))
+  };
+
+  toast.loading("Renaming file...");
+  axios.put(`/api/cdn/resize/image`, JSON.stringify(data)).then((res) => {
+    if (res.status === 200) {
+      toast.dismiss();
+      toast.success("File resized!");
+
+      setTimeout(() => {
+        location.reload();
+      }, 1500)
+    }
+  }).catch((err) => {
+    toast.dismiss();
+    toast.error("Error: " + err.response.data);
+  })
+}
```

Comment:
toast.loading returns an id, you can take this id and pass it to success and error so you don't need to use toast.dismiss
```suggestion
  const toastId = toast.loading("Renaming file...");
  axios.put(`/api/cdn/resize/image`, JSON.stringify(data)).then((res) => {
    if (res.status === 200) {
      toast.success("File resized!", { id: toastId, duration: 4000 });
      
      setTimeout(() => {
        location.reload();
      }, 1500)
    }
  }).catch((err) => {
    toast.error("Error: " + err.response.data, { id: toastId, duration: 4000 });
  })
}
```